### ISSUE TITLE
Error after migrating from XML to MS SQL

### ISSUE THREAD

**Issue by Deltaworks** [Orckestra/C1-CMS-Foundation#306]
Hi,

I just started a new project.
Downloaded a clean install using WebMatrix.
Added the SQL Server Data Provider function to the site
Created a new database @ my hosting company
Did the  Convert to SQL step

And now I get the following error if try to run the site on my local machine:
**0 writeable data providers exists for data 'Composite.Data.Types.IUrlConfiguration'.**

I've tested the connection, i'm able to open my database using MS SQL Server Management Studio.

I read somewhere it might be because it needs dbo rights. I've asked my hosting company to give me dbo rights on my user, as they did.

Still getting the same error.

Can anyone help me resolve this?

Regards,

Alex

**Comment by burningice2866** [comment]
@Deltaworks Have you looked in the Server Log? its a txt file located in the App_Data/Composite/Logs folder which should be able to tell you whats wrong.

**Comment by mtlive** [comment]
I have another problem with this SQL migration.
I tried to convert my website to a Sql server based site. I got the success message from the add-on but when I refreshed my console I faced with Runtime Error.
 I could restore my website by renaming DataStores.bak to DataStores and DynamicXmlDataProvider.config.bak to DynamicXmlDataProvider.config. Also database tables are created.
 Why it happened? Is there anything else I should do to cleanup the failed conversion?

 Part of the Log:

 20160803 10:59:18.4022 4 143 Critical XmlDataProvider  Failed to save data to the file file:D:\HostingSpaces\posat\posattt.com\wwwroot\App_Data\Composite\DataStores\Composite.Data.Types.IUserConsoleInformation.xml
 20160803 10:59:18.4022 4 143 Critical XmlDataProvider  System.IO.DirectoryNotFoundException: Could not find a part of the path 'D:\HostingSpaces\posat\posattt.com\wwwroot\App_Data\Composite\DataStores\Composite.Data.Types.IUserConsoleInformation.xml.rd3bvxdm.tmp'.
 at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
 at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy, Boolean useLongPath, Boolean checkHost)
 at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options, String msgPath, Boolean bFromProxy)
 at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, Boolean useAsync)
 at System.Xml.XmlWriterSettings.CreateWriter(String outputFileName)
 at System.Xml.XmlWriter.Create(String outputFileName, XmlWriterSettings settings)
 at Composite.Plugins.Data.DataProviders.XmlDataProvider.Foundation.XmlDataProviderDocumentWriter.DoSave(FileRecord fileRecord)
